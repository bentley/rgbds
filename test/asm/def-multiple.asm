def w = 1, x = w+1, y = x+1
	println "{d:w} {d:x} {d:y}"
def w set y+2, x set w+2, y set x+2,
	println "{d:w} {d:x} {d:y}"
redef w = w*2, x = w*2, y = x*2,
	println "{d:w} {d:x} {d:y}"
redef w set w-5, x set w-5, y set x-5
	println "{d:w} {d:x} {d:y}"

DEF spam EQU 6, eggs EQU 7, liff EQU spam*eggs
	println "{d:spam} x {d:eggs} = {d:liff}"

DEF true EQUS "no", false EQUS "yes", filenotfound EQUS "maybe"
	println "{true} or {false} (or {filenotfound})"
REDEF true EQUS "{false}", false EQUS "{true}" ; oops, this isn't Python!
	println "{true} xor {false}"

def toast EQU 1, spam EQU 2, lobster EQU 3 ; error, 'spam' already defined
	println "{d:toast}, {d:spam}, {d:lobster}"

def cannot = 10, mix EQU 20, types EQUS "30" ; syntax error
