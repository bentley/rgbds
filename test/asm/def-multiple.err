ERROR: def-multiple.asm(18):
    'spam' already defined at def-multiple.asm(10)
ERROR: def-multiple.asm(21):
    syntax error, unexpected EQU, expecting SET or =
error: Assembly aborted (2 errors)!
