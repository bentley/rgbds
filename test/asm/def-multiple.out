1 2 3
5 7 9
10 20 40
5 0 -5
6 x 7 = 42
no or yes (or maybe)
yes xor yes
1, 6, 3
